piton/tools/perlmod/Sims/1.12/man/man3/Sims.3
.\" Automatically generated by Pod::Man v1.34, Pod::Parser v1.13
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sh \" Subsection heading
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  | will give a
.\" real vertical bar.  \*(C+ will give a nicer C++.  Capital omega is used to
.\" do unbreakable dashes and therefore won't be available.  \*(C` and \*(C'
.\" expand to `' in nroff, nothing in troff, for use with C<>.
.tr \(*W-|\(bv\*(Tr
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.Sh), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.if \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.\"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.hy 0
.if n .na
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "Sims 3"
.TH Sims 3 "2005-05-05" "perl v5.8.0" "User Contributed Perl Documentation"
.SH "NAME"
Sims \- Perl module to configure sims for a given project
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
use Sims;
.SH "ABSTRACT"
.IX Header "ABSTRACT"
Exports project-specific variables to \fBsims\fR simulation script.
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
A hash called \f(CW%proj_vars\fR is exported, which has the following fields:
.PP
.Vb 6
\&        proj            the name of the project, OpenSPARCT1 or OpenPiton
\&        sims_config     directory of config files for sys=<foo> argument
\&        diaglist        path to top of diag directory
\&        sas             name of architectural simulator
\&        sasconfig       name of program to generate sas config file
\&        sims_db         list of directories where big-brother logs are kept
.Ve
.PP
Paths are relative to \fBdv_root\fR.
The values in \f(CW%proj_vars\fR depend on the environment variable \fB$PROJECT\fR.
.Sh "\s-1EXPORT\s0"
.IX Subsection "EXPORT"
\&':all'	exports the hash \f(CW%proj_vars\fR.
No default export.
.SH "SEE ALSO"
.IX Header "SEE ALSO"
\&\*(L"sims \-h\*(R"
.PP
This pathetic library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself. 


piton/tools/perlmod/Sims/1.12/lib/site_perl/5.8.0/Sims.pm
# Modified by Princeton University on June 9th, 2015
# ========== Copyright Header Begin ==========================================
# 
# OpenSPARC T1 Processor File: Sims.pm
# Copyright (c) 2006 Sun Microsystems, Inc.  All Rights Reserved.
# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES.
# 
# The above named program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public
# License version 2 as published by the Free Software Foundation.
# 
# The above named program is distributed in the hope that it will be 
# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# General Public License for more details.
# 
# You should have received a copy of the GNU General Public
# License along with this work; if not, write to the Free Software
# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
# 
# ========== Copyright Header End ============================================
package Sims;

use 5.008;
use strict;
use warnings;

require Exporter;

our @ISA = qw(Exporter);

# Items to export into callers namespace by default. Note: do not export
# names by default without a very good reason. Use EXPORT_OK instead.
# Do not simply export all your public functions/methods/constants.

# This allows declaration	use Sims ':all';
# If you do not need this, moving things directly into @EXPORT or @EXPORT_OK
# will save memory.

## Default project is BW.  Perhaps it should not be set.  [Viranjit 3/29/04]
## mmckeown: Changing to OpenPiton by default
my $project = exists $ENV{PROJECT} ? $ENV{PROJECT} : "OpenPiton";
our %proj_vars;

if ( $project eq 'OpenSPARCT1' ) {
    %proj_vars = (
      'proj'		=> 'OpenSPARCT1',
      'sims_config'	=> 'tools/src/sims',
      'diaglist'	=> 'verif/diag',
      'sas'		=> 'bas',
      'sasconfig'	=> 'basconfig',
      'sims_db'		=> [],
      'vcsprefix'       => '',
      'cc_dv_root'      => '',
      'has_denali'  => '0'
    ) ;
} elsif ( $project eq 'OpenPiton' ) {
    %proj_vars = (
      'proj'        => 'OpenPiton',
      'sims_config' => 'tools/src/sims',
      'diaglist'    => 'verif/diag',
      'sas'     => 'bas',
      'sasconfig'   => 'basconfig',
      'sims_db'     => [],
      'vcsprefix'       => '',
      'cc_dv_root'      => '',
      'has_denali'  => '0'
    ) ;
} else {
    die   "Sims.pm: Unknown project: \$PROJECT = $ENV{PROJECT}," .
	"\n         only OpenSPARCT1 or OpenPiton is supported, please do one of the following:" .
	"\n         setenv PROJECT OpenSPARCT1\nOR\n         setenv PROJECT OpenPiton\nOR\n         unsetenv PROJECT\n"
;
}

our %EXPORT_TAGS = ( 'all' => [ qw( %proj_vars
	
) ] );

our @EXPORT_OK = ( @{ $EXPORT_TAGS{'all'} } );

our @EXPORT = qw(
	
);

our $VERSION = '1.12';


# Preloaded methods go here.

1;
__END__

=head1 NAME

Sims - Perl module to configure sims for a given project

=head1 SYNOPSIS

use Sims;

=head1 ABSTRACT

Exports project-specific variables to B<sims> simulation script.

=head1 DESCRIPTION

A hash called %proj_vars is exported, which has the following fields:

	proj		the name of the project, OpenSPARCT1 or OpenPiton
	sims_config	directory of config files for sys=<foo> argument
	diaglist	path to top of diag directory
	sas		name of architectural simulator
	sasconfig	name of program to generate sas config file
	sims_db		list of directories where big-brother logs are kept

Paths are relative to B<dv_root>.
The values in %proj_vars depend on the environment variable B<$PROJECT>.

=head2 EXPORT

':all'	exports the hash %proj_vars.
No default export.

=head1 SEE ALSO

L<sims -h>

=cut
